src/datajudge/constraints/uniques.py

```diff
@@ -292,3 +294,76 @@ def compare(
     def test(self, engine: sa.engine.Engine) -> TestResult:
         self.max_relative_gain = self.max_relative_gain_getter(engine)
         return super().test(engine)
+
+
+class VariantDistributionConstraint(Constraint):
```

Comment:
I am not a 100% confident in the name but don't have a lot of other ideas as well.
Something along the line of `categorical` or `factor` would help one to understand it more directly, probably.

Reply:
I agree.
What about `CategoricalBoundConstraint`? It doesn't roll beautifully off the tongue, either, though. ;)

If I understand correctly, we don't need to always specify an entire distribution. E.g. if there are distinct values A, B, C, D, E and F, we could also use this constraint - and not just for the sake of the argument, but with real value - to bound the shares of A, B, C, D - which together we assume to cover 99% of the rows. We could then set `default_bound=(0, .01)`.